quinn/src/runtime.rs

```diff
@@ -84,7 +85,11 @@ pub fn default_runtime() -> Option<Arc<dyn Runtime>> {
         return Some(Arc::new(AsyncStdRuntime));
     }
 
-    #[cfg(not(feature = "runtime-async-std"))]
+    #[cfg(feature = "runtime-smol")]
+    {
+        return Some(Arc::new(SmolRuntime));
```

Comment:
This, and the `None` clause below, produce unreachable code warnings in `cargo check --all-features` if `runtime-async-std` is enabled. Please adjust feature gates such that `cargo check --all-features` is clean, e.g. by using `#[cfg(all(feature = "runtime-smol", not(feature = "runtime-async-std")))]` here, and `#[cfg(not(any(feature = "runtime-async-std", feature = "runtime-smol")))]` on the `None` case.